ssh/ssh_tunnel.py

```diff
@@ -0,0 +1,138 @@
+"""
+Module for creating persistent SSH connections for use with synchronous
+commands. Typically, tunnels should be invoked with a context manager to
+ensure proper cleanup. E.G.:
+with contextlib.closing(SSHTunnel(*args, **kwargs)) as tunnel:
+    ssh.write_to_remote('/usr/local/usrpath/testfile.txt', 'test_file.txt')
+    ssh.remote_cmd(['cat', 'test_file.txt'])
+"""
+import logging
+from contextlib import closing
+from subprocess import CalledProcessError, TimeoutExpired, check_call, check_output
+
+logger = logging.getLogger(__name__)
+
+SOCKET_DIR = '$HOME/.cache/ssh'
+SSH_CMD = [
+    '/usr/bin/ssh',
+    '-oConnectTimeout=10',
+    '-oControlMaster=auto',
```

Comment:
Code wise this is duplicating stuff that is in the async ssh module. Can we make it not be duplicated? 

Is there any particular reason why we don't want to turn on using managed connections for all of the installer SSH Library? Seems like a good global optimization